packages/drivers/routerlicious-driver/src/queryStringUtils.ts

```diff
@@ -0,0 +1,21 @@
+export type QueryStringType = Record<string, string | number | boolean>;
+
+/**
+ * Generates query string from the given query parameters.
+ * @param queryParams - Query parameters from which to create a query.
+ * @param url - URL to which query params should be appended
+ */
+export function buildQueryString(queryParams: QueryStringType, url?: string): string {
```

Comment:
nit:reading through the logic where this is used, I got a bit confused because the `buildQueryString` name implies (to me) that this will just return a QueryString, not a URL with a QueryString. I also can't see anywhere that URL is not passed in.

Also, I'd recommend using the [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) API here instead of manually creating and encoding the query string.

Example,

```typescript
/**
 * Generates URL string from the given URL and query parameters.
 * @param url - URL to which query params should be appended. Can include base/default query params.
 * @param queryParams - Query parameters from which to create a query. Will override any query params in url.
 */
export function buildUrlWithQueryString(url: string, queryParams: QueryStringType): string {
  // Initialize urlSearchParams with query params from the base URL itself
  const urlSearchParams = new URLSearchParams(url);
  for (const [key, value] of Object.entries(queryParams)) {
    // Add/override search params from query params
    urlSearchParams.set(key, value);
  }
  return `${url.split("?")[0]}?${urlSearchParams.toString()}`;
}
```